### PR TITLE
Upgradle to 4.0

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

File: gradle/java.gradle [hunk 1]
```diff
@@ -6,6 +6,10 @@ apply plugin: 'com.palantir.baseline-idea'
 
 sourceCompatibility = 1.8
 
+checkstyle {
+    configDir = rootProject.file(".baseline")
```
Comment (robert3005) [inline gradle/java.gradle:10]: necessary until gradle-baseline picks up new gradle-baseline

File: build.gradle [hunk 2]
```diff
@@ -8,15 +8,15 @@ buildscript {
     dependencies {
         classpath 'com.gradle.publish:plugin-publish-plugin:0.9.1'
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.6'
-        classpath 'com.netflix.nebula:nebula-dependency-recommender:4.0.2'
-        classpath 'com.netflix.nebula:nebula-publishing-plugin:4.9.1'
+        classpath 'com.netflix.nebula:nebula-dependency-recommender:4.2.1'
+        classpath 'com.netflix.nebula:nebula-publishing-plugin:5.1.0'
```
Comment (uschi2000) [inline build.gradle:12]: can you check that the produced POMs have versions populated, please?

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineCheckstyle.groovy [hunk 3]
```diff
@@ -54,14 +54,9 @@ class BaselineCheckstyle extends AbstractBaselinePlugin {
     def configureCheckstyle() {
         project.logger.debug("Baseline: Configuring Checkstyle tasks")
 
-        def configProps = project.checkstyle.configProperties
-        // Required to enable checkstyle suppressions
-        configProps['samedir'] = "${configDir}/checkstyle"
```
Comment (uschi2000) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineCheckstyle.groovy:59]: does this still work with eclipse and intellij?

File: gradle/java.gradle (hunk 1 above)
Comment (dansanduleac) [inline gradle/java.gradle]: `.baseline/checkstyle`?

File: build.gradle (hunk 2 above)
Comment (robert3005) [reply build.gradle:12]: Checked on local publish. I bumped these since without the bump versions were missing

File: gradle/java.gradle (hunk 1 above)
Comment (robert3005) [reply gradle/java.gradle:10]: Current baseline already sets configFile to `configDir/checkstyle/checkstyle.xml` and suppressions directory to `configDir/checkstyle`. My claim is when you release this and pick up here you would be able to remove this line

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineCheckstyle.groovy (hunk 3 above)
Comment (robert3005) [reply gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineCheckstyle.groovy:59]: Works in intellij, but don't have working eclipse setup to confirm

**Comment by robert3005** [comment]
also - this will require gradle 4.0 since configDir on checkstyle plugin was previously read only

**Comment by uschi2000** [comment]
yep

**Comment by uschi2000** [comment]
@robert3005 something is buggy with this patch, unfortunately. can you try upgrading one of the projects to see what's going on?

**Comment by robert3005** [comment]
submitted #188 that should cover the fix

**Pull request merged**